music/src/core/sequences.ts

```diff
@@ -325,3 +334,58 @@ export function assertIsAbsoluteQuantizedSequence(ns: INoteSequence) {
         ns.id} is not quantized or is quantized based on relative timing`);
   }
 }
+
+/**
+ * Create an unquantized version of a quantized `NoteSequence`.
+ *
+ * Any existing times will be replaced in the output `NoteSequence` and
+ * quantization info and steps will be removed.
+ *
+ * @param ns The `NoteSequence` to unquantize.
+ * @param qpm The tempo to use. If not provided, the tempo in `ns` is used,
+ * or the default of 120 if it is not specified in the sequence either.
+ * @returns a new non-quantized `NoteSequence` wih time in seconds.
+ */
+export function unquantizeSequence(qns: INoteSequence, qpm?: number) {
+  // TODO(adarob): Support absolute quantized times and multiple tempos.
+  assertIsRelativeQuantizedSequence(qns);
+  assertSingleTempo(qns);
+
+  const ns = clone(qns);
+
+  if (qpm) {
+    if (ns.tempos && ns.tempos.length > 0) {
+      ns.tempos[0].qpm = qpm;
+    } else {
+      ns.tempos.push(NoteSequence.Tempo.create({time: 0, qpm: qpm}));
+    }
+  } else {
+    qpm = (qns.tempos && qns.tempos.length > 0) ?
+        ns.tempos[0].qpm :
+        constants.DEFAULT_QUARTERS_PER_MINUTE;
+  }
+
+  const stepToSeconds = (step: number) =>
+      step / ns.quantizationInfo.stepsPerQuarter * (60 / qpm);
+  ns.totalTime = stepToSeconds(ns.totalQuantizedSteps);
+  ns.totalQuantizedSteps = undefined;
```

Comment:
I don't think we need to clear out quantized information when adding unquantized information. Having both is only problematic if we're trying to do something like concatenate quantized sequences.

Reply:
Done.